autode/path/adaptive.py

```diff
@@ -1,20 +1,29 @@
 import autode as ade
 import numpy as np
-from copy import deepcopy
+
 from autode.log import logger
-from autode.mol_graphs import make_graph
 from autode.path.path import Path
 from autode.transition_states.ts_guess import TSguess
 from autode.utils import work_in
+from autode.bonds import ScannedBond
+
+from typing import TYPE_CHECKING, List, Optional
+
+
+if TYPE_CHECKING:
+    from autode.species import ReactantComplex, ProductComplex, Species
+    from autode.transition_states import TSguess
```

Comment:
I think this might be duplicate of `from autode.transition_states.ts_guess import TSguess` from above?